iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -658,6 +929,141 @@ TEST_F(PoshRuntime_test, ShutdownUnblocksBlockingPublisher)
+    std::this_thread::sleep_for(std::chrono::milliseconds(SLEEP_IN_MS));
+    EXPECT_THAT(wasRequestSent.load(), Eq(false));
+
+    m_runtime->shutdown();
+
+    blockingServer.join(); // ensure the wasRequestSent store happens before the read
+    EXPECT_THAT(wasRequestSent.load(), Eq(true));
+}
+
+TEST_F(PoshRuntime_test, ShutdownUnblocksBlockingServer)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "f67db1c5-8db9-4798-b73c-7175255c90fd");
+    // get publisher and subscriber
+    iox::capro::ServiceDescription serviceDescription{"don't", "stop", "me"};
+
+    iox::popo::ClientOptions clientOptions;
+    clientOptions.responseQueueCapacity = 1U;
+    clientOptions.responseQueueFullPolicy = iox::popo::QueueFullPolicy::BLOCK_PRODUCER;
+    clientOptions.serverTooSlowPolicy = iox::popo::ConsumerTooSlowPolicy::WAIT_FOR_CONSUMER;
+
+    iox::popo::ServerOptions serverOptions;
+    serverOptions.requestQueueCapacity = 10U;
+    serverOptions.requestQueueFullPolicy = iox::popo::QueueFullPolicy::BLOCK_PRODUCER;
+    serverOptions.clientTooSlowPolicy = iox::popo::ConsumerTooSlowPolicy::WAIT_FOR_CONSUMER;
+
+    iox::popo::UntypedClient client{serviceDescription, clientOptions};
+    iox::popo::UntypedServer server{serviceDescription, serverOptions};
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(300));
+
+    ASSERT_TRUE(server.hasClients());
+    ASSERT_THAT(client.getConnectionState(), Eq(iox::ConnectionState::CONNECTED));
+
+    // send requests to fill request queue
+    for (uint64_t i = 0; i < clientOptions.responseQueueCapacity + 1; ++i)
+    {
+        auto clientLoanResult = client.loan(sizeof(uint64_t), alignof(uint64_t));
+        ASSERT_FALSE(clientLoanResult.has_error());
+        client.send(clientLoanResult.value());
+    }
+
+    auto threadSyncSemaphore = iox::posix::Semaphore::create(iox::posix::CreateUnnamedSingleProcessSemaphore, 0U);
+    std::atomic_bool wasResponseSent{false};
+
+    constexpr iox::units::Duration DEADLOCK_TIMEOUT{5_s};
+    Watchdog deadlockWatchdog{DEADLOCK_TIMEOUT};
+    deadlockWatchdog.watchAndActOnFailure([] { std::terminate(); });
+
+    // block in a separate thread
+    std::thread blockingServer([&] {
+        auto processRequest = [&]() {
+            auto takeResult = server.take();
+            ASSERT_FALSE(takeResult.has_error());
+            auto loanResult = server.loan(
+                iox::popo::RequestHeader::fromPayload(takeResult.value()), sizeof(uint64_t), alignof(uint64_t));
+            ASSERT_FALSE(loanResult.has_error());
+            server.send(loanResult.value());
+        };
+
+        for (uint64_t i = 0; i < clientOptions.responseQueueCapacity; ++i)
+        {
+            processRequest();
+        }
+
+        ASSERT_FALSE(threadSyncSemaphore->post().has_error());
+        processRequest();
+        wasResponseSent = true;
+    });
+
+    // wait some time to check if the server is blocked
+    constexpr int64_t SLEEP_IN_MS = 100;
```

Comment:
Same nitpick here, think about encoding the SI unit in the type instead of the variable name or move it to line 1058